test/x509_time_test.c

```diff
@@ -25,6 +25,46 @@ typedef struct {
     int expected;
 } TESTDATA;
 
+/*
+ * Actually, the "loose" mode has been tested in
+ * those time-compare-cases, so we may not test it again.
+ */
+static TESTDATA x509_strict_tests[] = {
+    {
+        "20170217180154Z", V_ASN1_GENERALIZEDTIME,
+        0, 1,
```

Comment:
I think you should probably either add something to the TESTDATA structure or use a different one. The current expected is the result of a compare, so you have a cmp_time and when comparing the parsed and the expected value you expect to get something. You don't use the cmp_time, and use expected for something else.

You also don't seem to be using the type you set in the test data, and adjust the type based on the string you give, which doesn't make sense. I think you should check both that parsing is successful and that it returned the correct type. The correct type for all of your strings should be V_ASN1_UTCTIME, because they're all before 2050. You should also create strings that are after 2050.